src/app/inventory/StoreBucket.tsx

```diff
@@ -33,23 +33,60 @@ interface ProvidedProps {
 
 // Props from Redux via mapStateToProps
 interface StoreProps {
+  destinyVersion: DestinyVersion;
+  storeId: string;
+  storeName: string;
+  storeClassType: DestinyClass;
+  isVault: boolean;
   items: DimItem[];
   itemSortOrder: string[];
-  allStores: DimStore[];
+  storeClassList: DestinyClass[];
   characterOrder: string;
   isPhonePortrait: boolean;
 }
 
-function mapStateToProps(state: RootState, props: ProvidedProps): StoreProps {
-  const { store, bucket } = props;
+/**
+ * Generate a function that will produce a new array (by-ref) only if the array's contents change.
+ */
+function makeInternArray<T>() {
```

Comment:
huh, cool! would this be useful elsewhere?

Reply:
Yeah, I'll probably move it to utils, though we shouldn't need it once some other stuff lands.